Analysis/EventFiltering/nucleiFilter.cxx

```diff
@@ -0,0 +1,139 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+// O2 includes
+
+#include "ReconstructionDataFormats/Track.h"
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoAHelpers.h"
+#include "AnalysisDataModel/PID/PIDResponse.h"
+#include "AnalysisDataModel/TrackSelectionTables.h"
+
+#include "AnalysisDataModel/EventSelection.h"
+#include "filterTables.h"
+
+#include "Framework/HistogramRegistry.h"
+
+#include <cmath>
+#include <string>
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+namespace
+{
+float rapidity(float pt, float eta, float m)
+{
+  return std::asinh(pt / std::hypot(m, pt) * std::sinh(eta));
+}
+
+static constexpr int nNuclei{4};
+static constexpr int nCutsPID{5};
+static constexpr std::array<float, nNuclei> masses{
+  constants::physics::MassDeuteron, constants::physics::MassTriton,
+  constants::physics::MassHelium3, constants::physics::MassAlpha};
+static constexpr std::array<int, nNuclei> charges{1, 1, 2, 2};
+static const std::vector<std::string> nucleiNames{"H2", "H3", "He3", "He4"};
+static const std::vector<std::string> cutsNames{
+  "TPCnSigmaMin", "TPCnSigmaMax", "TOFnSigmaMin", "TOFnSigmaMax", "TOFpidStartPt"};
+static constexpr float cutsPID[nNuclei][nCutsPID]{
```

Comment:
You can use a Configurable for this with a labelled matrix

Reply:
We are currenlty using a LabeledArray at line 67, do you mean this or something else?
https://github.com/AliceO2Group/AliceO2/pull/5967/files#diff-e9b9353f07d53b8f6356e36b02f270b629cf587994a542419780a5955a0b5340R67

Reply:
Yes, all perfect